cupyx/scipy/interpolate/_interpolate.py

```diff
@@ -869,3 +870,33 @@ def integrate(self, a, b, extrapolate=None):
         # Return
         range_int *= sign
         return range_int.reshape(self.c.shape[2:])
+
+    @classmethod
+    def from_spline(cls, tck, extrapolate=None):
+        """
+        Construct a piecewise polynomial from a spline
+
+        Parameters
+        ----------
+        tck
+            A spline, as a (knots, coefficients, degree) tuple or
+            a BSpline object.
+        extrapolate : bool or 'periodic', optional
+            If bool, determines whether to extrapolate to out-of-bounds points
+            based on first and last intervals, or to return NaNs.
+            If 'periodic', periodic extrapolation is used. Default is True.
+        """
+        if isinstance(tck, BSpline):
+            t, c, k = tck.tck
+            if extrapolate is None:
+                extrapolate = tck.extrapolate
+        else:
+            t, c, k = tck
```

Comment:
Minor: this can be simplified now that you're creating a BSpline object anyway. (In scipy, BSplines had to be retrofitted, hence this whole stanza)